swan/scripts/before-notebook.d/02_environment.sh

```diff
@@ -6,34 +6,42 @@
 # launched as regular user as it's this entity which is able to access 
 # eos and not the super user.
 
-if [ -n "${ROOT_LCG_VIEW_PATH}" ] && [ -n "${ROOT_LCG_VIEW_NAME}" ] && [ -n "${ROOT_LCG_VIEW_PLATFORM}" ];
-then
-  _log "Setting up CVMFS";
-  # Set environment for the Jupyter process
-  LOCAL_HOME=/home/$NB_USER
-  export IPYTHONDIR=$LOCAL_HOME/.ipython
-  export JPY_DIR=$LOCAL_HOME/.jupyter
-
-  # The Jupyter environment variables below are set to point to the local user
-  # home (/home/$USER), instead of the EOS path
-  export JUPYTER_CONFIG_DIR=$JPY_DIR
-  export JUPYTER_PATH=$LOCAL_HOME/.local/share/jupyter
-  export JUPYTER_RUNTIME_DIR=$JUPYTER_PATH/runtime
-  export JUPYTER_DATA_DIR=$JUPYTER_PATH
-
-  # Set npm cache on the local user home, instead of the EOS path
-  export NPM_CONFIG_CACHE=$LOCAL_HOME/.npm
-
-  # Set jupyterlab config directory
-  export JUPYTERLAB_SETTINGS_DIR=$HOME/.local/lab/user-settings/
-
-  # Set other environment variables
-  export KERNEL_DIR=$JUPYTER_PATH/kernels
-  export SWAN_ENV_FILE=$LOCAL_HOME/.bash_profile
-  export PROFILEPATH=$IPYTHONDIR/profile_default
-
-  # Create missing directories
-  mkdir -p $IPYTHONDIR $PROFILEPATH
+# Set environment for the Jupyter process
+LOCAL_HOME=/home/$NB_USER
+export IPYTHONDIR=$LOCAL_HOME/.ipython
+export JPY_DIR=$LOCAL_HOME/.jupyter
+
+# The Jupyter environment variables below are set to point to the local user
+# home (/home/$USER), instead of the EOS path
+export JUPYTER_CONFIG_DIR=$JPY_DIR
+export JUPYTER_PATH=$LOCAL_HOME/.local/share/jupyter
+export JUPYTER_RUNTIME_DIR=$JUPYTER_PATH/runtime
+export JUPYTER_DATA_DIR=$JUPYTER_PATH
+
+# Set npm cache on the local user home, instead of the EOS path
+export NPM_CONFIG_CACHE=$LOCAL_HOME/.npm
+
+# Set jupyterlab config directory
+export JUPYTERLAB_SETTINGS_DIR=$HOME/.local/lab/user-settings/
+
+# Set other environment variables
+export KERNEL_DIR=$JUPYTER_PATH/kernels
+export SWAN_ENV_FILE=$LOCAL_HOME/.bash_profile
+export PROFILEPATH=$IPYTHONDIR/profile_default
+
+# Create missing directories
+mkdir -p $IPYTHONDIR $PROFILEPATH
+
+# Make the user the owner of the local home and subdirectories
+chown -R $NB_USER:$NB_GID $LOCAL_HOME
+
+# Create a gitignore file for ignoring all .sys. files and set it as global
+# This will hide the checkpoint files created by EOS.
+GLOBAL_GITIGNORE="$LOCAL_HOME/.gitignore_global"
+echo ".sys.*" > "$GLOBAL_GITIGNORE"
+run_as_user git config --global core.excludesfile "$GLOBAL_GITIGNORE"
```

Comment:
This should be placed before the chown above, otherwise the file won't have the same ownership as the rest of the files in the directory:
```
-rw-r--r--.  1 root     users    7 May  7 11:45 .gitignore_global
```